internal/services/mssql/mssql_virtual_machine_resource.go

```diff
@@ -811,7 +811,7 @@ func flattenSqlVirtualMachineStorageConfigurationSettings(input *sqlvirtualmachi
 		"disk_type":             string(input.DiskConfigurationType),
 		"data_settings":         flattenSqlVirtualMachineStorageSettings(input.SQLDataSettings),
 		"log_settings":          flattenSqlVirtualMachineStorageSettings(input.SQLLogSettings),
-		"temp_db_settings":      flattenSqlVirtualMachineStorageSettings(input.SQLTempDbSettings),
+		"temp_db_settings":      flattenSqlVirtualMachineTempDbSettings(input.SQLTempDbSettings),
```

Comment:
The type has been changed in the new api version so created a copy of the old expand/flatten